contrib/test_jenkins.sh

```diff
@@ -662,73 +701,67 @@ run_uct_hello() {
 }
 
 run_client_server() {
+	test_name=ucp_client_server
 
-    test_name=ucp_client_server
-
-    if [ ! -x ${test_name} ]
-    then
-        gcc -o ${test_name} ${ucx_inst}/share/ucx/examples/${test_name}.c \
-            -lucp -lucs -I${ucx_inst}/include -L${ucx_inst}/lib \
-            -Wl,-rpath=${ucx_inst}/lib
-    fi
-
-    iface=`ibdev2netdev | grep Up | awk '{print $5}' | head -1`
-    if [ -n "$iface" ]
-    then
-        server_ip=`ip addr show ${iface} | awk '/inet /{print $2}' | awk -F '/' '{print $1}'`
-    fi
-
-    if [ -z "$server_ip" ]
-    then
-        # if there is no inet (IPv4) address, bail
-        return
-    fi
-
-    ibdev=`ibdev2netdev | grep $iface | awk '{print $1}'`
-    node_guid=`cat /sys/class/infiniband/$ibdev/node_guid`
-    if [ $node_guid == "0000:0000:0000:0000" ]
-    then
-        return
-    fi
+	if [ ! -x ${test_name} ]
+	then
+		gcc -o ${test_name} ${ucx_inst}/share/ucx/examples/${test_name}.c \
+			-lucp -lucs -I${ucx_inst}/include -L${ucx_inst}/lib \
+			-Wl,-rpath=${ucx_inst}/lib
+	fi
 
-    server_port=$((10000 + EXECUTOR_NUMBER))
+	iface=`ibdev2netdev | grep Up | awk '{print $5}' | head -1`
+	if [ -n "$iface" ]
+	then
+		server_ip=`ip addr show ${iface} | awk '/inet /{print $2}' | awk -F '/' '{print $1}'`
+	fi
 
-    # run server side
-    ./${test_name} -p ${server_port} &
-    hw_server_pid=$!
+	if [ -z "$server_ip" ]
+	then
+		# if there is no inet (IPv4) address, bail
+		return
+	fi
 
-    sleep 15
+	ibdev=`ibdev2netdev | grep $iface | awk '{print $1}'`
+	node_guid=`cat /sys/class/infiniband/$ibdev/node_guid`
```

Comment:
maybe add helper function is_rdma_device ${iface}? it will unite the checks in line 713 and 727

Reply:
> maybe add helper function is_rdma_device ${iface}? it will unite the checks in line 713 and 727

@yosefe not sure that I understand your idea
I've united all `713-727` lines to `get_ipoib_addr` function that gets first Up IPoIB address